src/vm/checker/typecheck/mod.rs

```diff
@@ -151,8 +151,8 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
             Some(ref mut tracker) => {
                 let new_type = match tracker.take() {
                     Some(expected_type) => {
-                        TypeSignature::most_admissive(expected_type, return_type)
-                            .map_err(|_| CheckError::new(CheckErrors::ReturnTypesMustMatch))?
+                        TypeSignature::most_admissive(expected_type.clone(), return_type.clone())
+                            .map_err(|_| CheckError::new(CheckErrors::ReturnTypesMustMatch(expected_type, return_type)))?
```

Comment:
You can eliminate these clones in `map_err` by handling the err type of `most_admissive` (this is why the err case returns the two types):

```rust
                       TypeSignature::most_admissive(expected_type, return_type)
                            .map_err(|(a, b)| CheckError::new(CheckErrors::ReturnTypesMustMatch(a, b)))?
```

Reply:
Brilliant, thanks.
Addressed with https://github.com/blockstack/blockstack-core/pull/1041/commits/dfacb432b383eb88f9c6df483e54466f3d29bf3e